gradio/deploy_space.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+import json
+import os
+import random
+import re
+
+import huggingface_hub
+
+
+def add_configuration_to_readme(repo_directory, readme_file) -> dict:
+    configuration = {}
+
+    dir_name = os.path.basename(repo_directory)
+    title = input(f"Enter Spaces app title [{dir_name}]: ") or dir_name
+    formatted_title = format_title(title)
+    if formatted_title != title:
+        print(f"Formatted to {formatted_title}. ")
+    configuration["title"] = formatted_title
+
+    app_file = None
+    for file in os.listdir(repo_directory):
+        file_path = os.path.join(repo_directory, file)
+        if (not os.path.isfile(file_path)) or (
+            not file.endswith(".py") and not file.endswith(".ipynb")
+        ):
+            continue
+
+        with open(file_path, encoding="utf-8", errors="ignore") as f:
+            content = f.read()
+            if "import gradio" in content:
+                app_file = file
+                break
+
+    app_file = (
+        input(f"Enter Gradio app file {f'[{app_file}]' if app_file else ''}: ")
+        or app_file
+    )
+    if not app_file or not os.path.exists(app_file):
+        raise FileNotFoundError("Failed to find Gradio app file.")
+    if app_file.endswith(".ipynb"):
+        configuration["notebook_file"] = app_file
+        app_file = "_" + configuration["notebook_file"].replace(".ipynb", ".py")
+    configuration["app_file"] = app_file
+
+    emoji_set = "🤯🤖🧠🐶👑💥🐮🎁🐙🦋"
+    default_emoji = random.choice(emoji_set)
+    configuration["emoji"] = (
+        input(f"Enter Spaces Card emoji [{default_emoji}]: ") or default_emoji
+    )
+
+    color_set = ["red", "yellow", "green", "blue", "indigo", "purple", "pink", "gray"]
+    default_color = random.choice(color_set)
+    color = input(f"Enter Spaces Card color [{default_color}]: ") or default_color
+    configuration["colorFrom"] = color
```

Comment:
Would be nice to add sdk_version and python_version as well.

Reply:
Done.